File: ios/SpentyAI/SpentyAI/Core/Models/Mandate.swift
import Foundation

struct Mandate: Codable, Identifiable {
    let id: String
    var merchant: String?
    var mandateType: String?
    var amount: Double?
    var frequency: String?
    var startDate: Date?
    var status: String?
    var source: String?
    var sourceEmailId: String?
    var sourceEmailSubject: String?

    enum CodingKeys: String, CodingKey {
        case id = "mandateId"
        case merchant
        case mandateType
        case amount
        case frequency
        case startDate
        case status
        case source
        case sourceEmailId
        case sourceEmailSubject
    }
}

File: android-native/app/src/main/java/com/spentyai/app/core/models/Mandate.kt
package com.spentyai.app.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Mandate(
    @SerialName("mandateId") val id: String = "",
    val merchant: String? = null,
    val mandateType: String? = null,
    val amount: Double? = null,
    val frequency: String? = null,
    val startDate: String? = null,
    val status: String? = null,
    val source: String? = null,
    val sourceEmailId: String? = null,
    val sourceEmailSubject: String? = null
)
